### PR TITLE
Ei 535 ei 490 ei 226 data locks acceptance tests

### PR DESCRIPTION
**Pull request by ruslanss**

### CODE REVIEW

File: src/tests/SFA.DAS.EmployerIncentives.Functions.PaymentsProcess.AcceptanceTests/Bindings/DatabasePerScenarioHook.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        public void CreateDatabase(TestContext context)
+        {
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
+            context.SqlDatabase = new SqlDatabaseForTest();
+            stopwatch.Stop();
+            Console.WriteLine($"[{nameof(DatabasePerScenarioHook)}] time it took to deploy test database: {stopwatch.Elapsed.Seconds} seconds");
+        }
+
+        [BeforeScenario]
```
Comment (Stephek) [inline src/tests/SFA.DAS.EmployerIncentives.Functions.PaymentsProcess.AcceptanceTests/Bindings/DatabasePerScenarioHook.cs:21]: AfterScenario ?

File: src/tests/SFA.DAS.EmployerIncentives.Functions.PaymentsProcess.AcceptanceTests/Services/TestPaymentsProcessFunctions.cs [hunk 2]
```diff
@@ -43,163 +47,107 @@ public void Dispose()
...
-            return await CompleteFunctionOrchestration(orchestrationLinks);
+            var orchestrationLinks = await StartIncentivePaymentOrchestrator(collectionPeriodYear, collectionPeriodMonth);
+            return await FunctionOrchestrationCompleted(orchestrationLinks);
         }
 
         public async Task StartLearnerMatching()
         {
             await StartLearnerMatchingOrchestrator();
-            Thread.Sleep(TimeSpan.FromSeconds(10));
+            Thread.Sleep(TimeSpan.FromSeconds(1)); // time it takes function host to update storage 🤷‍
```
Comment (Stephek) [inline src/tests/SFA.DAS.EmployerIncentives.Functions.PaymentsProcess.AcceptanceTests/Services/TestPaymentsProcessFunctions.cs:64]: Can we ping the health check instead to it's not a random thread.sleep?

File: src/SFA.DAS.EmployerIncentives.Data.UnitTests/TestHelpers/SqlDatabaseForTest.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.IO;
+
+namespace SFA.DAS.EmployerIncentives.Data.UnitTests.TestHelpers
+{
+    public class SqlDatabaseForTest : IDisposable
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Data.UnitTests/TestHelpers/SqlDatabaseForTest.cs:8]: It's implicit from the namespace that it's for test so maybe just call it SqlDatabase.

**Review by Stephek — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.EmployerIncentives.Data.UnitTests/TestHelpers/SqlDatabaseForTest.cs (hunk 3 above)
Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Data.UnitTests/TestHelpers/SqlDatabaseForTest.cs:8]: will do after i replace the original

File: src/tests/SFA.DAS.EmployerIncentives.Functions.PaymentsProcess.AcceptanceTests/Bindings/DatabasePerScenarioHook.cs (hunk 1 above)
Comment (ruslanss) [reply src/tests/SFA.DAS.EmployerIncentives.Functions.PaymentsProcess.AcceptanceTests/Bindings/DatabasePerScenarioHook.cs:21]: ooooohh good spot!

File: src/tests/SFA.DAS.EmployerIncentives.Functions.PaymentsProcess.AcceptanceTests/Services/TestPaymentsProcessFunctions.cs (hunk 2 above)
Comment (ruslanss) [reply src/tests/SFA.DAS.EmployerIncentives.Functions.PaymentsProcess.AcceptanceTests/Services/TestPaymentsProcessFunctions.cs:64]: we *are* using status API (see AllFunctionOrchestrationCompleted) however, it takes a bit of time for a function run to show up in there apparently (in any status) . so this is a necessity unfortunately.

**Pull request merged**